packages/driver/src/util/serialization/log.ts

```diff
@@ -0,0 +1,391 @@
+import _ from 'lodash'
+import { isSerializableInCurrentBrowser, preprocessForSerialization } from './index'
+import $dom from '../../dom'
+
+interface PreprocessedHTMLElement {
+  value?: any
+  type?: any
+  id?: string
+  tagName: string
+  attributes: { [key: string]: string }
+  innerHTML: string
+  serializationKey: 'dom'
+}
+
+interface PreprocessedFunction {
+  value: any
+  serializationKey: 'function'
+}
+
+/**
+ * Takes an HTMLElement that might be a <body> for a given snapshot or any other element, likely pertaining to log consoleProps,
+ * on the page that needs to be preprocessed for serialization. The method here is to do a very shallow serialization,
+ * by trying to make the HTML as stateful as possible before preprocessing.
+ *
+ * @param {HTMLElement} props - an HTMLElement
+ * @returns {PreprocessedHTMLElement} a preprocessed element that can be fed through postMessage() that can be reified in the primary.
+ */
+const preprocessDomElement = (props: HTMLElement) => {
```

Comment:
right now for `log`s and `snapshots` DOM elements are attempted to be serialized by their `innerHTML` as well as a few other properties to capture element state. This allows for some state to be captured when the elements are reified in the primary.